lib/index.js

```diff
@@ -290,7 +326,7 @@ util.inherits(Device, EventEmitter);
  */
 function createSocket (type, opts) {
     var domain = (opts || {}).raw ? nn.AF_SP_RAW : nn.AF_SP;
-    return new Socket(domain, type);
+    return new Socket(domain, type, opts);
```

Comment:
ya probably, sure. 

we definitely need to just pass opts, domain included, it's very ugly without doing that